pydarn/plotting/power.py

```diff
@@ -0,0 +1,182 @@
+                              "only higher frequencies will be plotted"
+                              "".format(frequency))
+                for record in low_freq_records:
+                    stat_pwr = statistical_calc(record['pwr0'])
+                    record.update({'pwr0': stat_pwr})
+                RTP.plot_time_series(low_freq_records, parameter='pwr0',
+                                     beam_num=beam_num)
+                plt.title(' Lag 0 Power for {} Beam: {} '.format(radar_abbrev,
+                                                                 beam_num))
+                plt.ylabel("{} Power\n [raw units]".format(statistical_calc))
+                plt.legend(["{} kHz".format(low_freq_records[0]['tfreq'])])
+                plt.ylabel("{} Power\n [raw units]".format(statistical_calc))
+            else:
+                for record in high_freq_records:
+                    stat_pwr = statistical_calc(record['pwr0'])
+                    record.update({'pwr0': stat_pwr})
+                for record in low_freq_records:
+                    stat_pwr = statistical_calc(record['pwr0'])
+                    record.update({'pwr0': stat_pwr})
+                plt.subplot(2, 1, 1)
+                RTP.plot_time_series(high_freq_records, parameter='pwr0',
+                                     beam_num=beam_num)
+                plt.title(' Lag 0 Power for {} Beam: {} '.format(radar_abbrev,
+                                                                 beam_num))
+                plt.ylabel("{} Power\n [raw units]".format(statistical_calc))
+                plt.legend(["{} kHz".format(high_freq_records[0]['tfreq'])])
+                plt.xticks([])
+                plt.subplot(2, 1, 2)
+                RTP.plot_time_series(low_freq_records, parameter='pwr0',
+                                     beam_num=beam_num)
+                plt.ylabel("{} Power\n [raw units]".format(statistical_calc))
+                plt.legend(["{} kHz".format(low_freq_records[0]['tfreq'])])
+        else:
+            # get records of interest that have a specific frequency
+            records_of_interest = [record for record in records
+                                   if record['tfreq'] == frequency]
+
+            if len(records_of_interest) == 0:
+                raise(exceptions.plot_exceptions.\
+                      NoDataFoundError('tfreq', beam_num,
+                                       opt_beam_num=records[0]['bmnum'],
+                                       opt_parameter_value=records[0]['tfreq']))
+
+            # gather important info regarding the records of interest
+            date = records_of_interest[0]['origin.time']
+            stid = records_of_interest[0]['stid']
+            radar_abbrev = SuperDARNRadars.radars[stid].hardware_info.abbrev
+
+            for record in records_of_interest:
+                stat_pwr = statistical_calc(record['pwr0'])
+                record.update({'pwr0': stat_pwr})
```

Comment:
Just wondering if this is a good idea to modify the pwr0 field in the record. If you were to use these records later then the pwr0 vector is no longer there since it's been modified. You would have to re-read the file. I think this might be a hidden behaviour to the user.

Reply:
since `records_of_interest` is a copy of records that have a frequency of interest so when we modify the pwr value here we actually are not modifying the original records the user passes in. Does that make sense? 

I tested this just to confirm.

Reply:
Hmm that's different from what I got, I did this and records[0] was modified. I think it's just a dictionary reference in the `records_of_interest` list. 
```
reader=pydarn.io.supdarn_io.SuperDARNRead('/home/marci/data/20201103.0001.00.inv.rawacf')
records=reader.read_rawacf()
print(records[0]['pwr0'])
pydarn.Power.plot_pwr0_statistic(records, frequency=10500, beam_num=2, compare=False)
print(records[0]['pwr0'])
```